scylla-cql/src/frame/value.rs

```diff
@@ -267,9 +288,39 @@ impl CqlVarint {
     }
 }
 
-impl CqlVarint {
+/// Conversion to bytes
+impl CqlVarintBorrowed<'_> {
+    /// Returns a slice of bytes in two's complement
+    /// binary big-endian representation.
+    pub fn as_signed_bytes_be_slice(&self) -> &[u8] {
+        self.0
+    }
+}
+
+/// An internal utility trait used to implement [`AsNormalizedVarintSlice`]
+/// for both [`CqlVarint`] and [`CqlVarintBorrowed`].
+trait AsVarintSlice {
+    fn as_slice(&self) -> &[u8];
+}
+impl AsVarintSlice for CqlVarint {
+    fn as_slice(&self) -> &[u8] {
+        self.as_signed_bytes_be_slice()
+    }
+}
+impl AsVarintSlice for CqlVarintBorrowed<'_> {
+    fn as_slice(&self) -> &[u8] {
+        self.as_signed_bytes_be_slice()
+    }
+}
+
+/// An internal utility trait used to implement [`PartialEq`] and [`std::hash::Hash`]
+/// for [`CqlVarint`] and [`CqlVarintBorrowed`].
+trait AsNormalizedVarintSlice {
+    fn as_normalized_slice(&self) -> &[u8];
+}
+impl<V: AsVarintSlice> AsNormalizedVarintSlice for V {
```

Comment:
This traits wouldn't be needed if `CqlVaring` were implemented using `Cow` inside.